File: test-2025.9.2/编程/排序/快速排序.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr
    else:
        pivot = arr[0]
        less = [x for x in arr[1:] if x <= pivot]
        greater = [x for x in arr[1:] if x > pivot]
        return quick_sort(less) + [pivot] + quick_sort(greater)

def main():
    n = int(input("请输入序列长度: "))
    sequence = list(map(int, input("请输入序列，以空格分隔: ").split()))
    sorted_sequence = quick_sort(sequence)
    print(" ".join(map(str, sorted_sequence)))

if __name__ == "__main__":
    main()

File: test-2025.9.2/编程/排序/快速排序.cpp
#include <iostream>
#include <vector>
#include <algorithm>

void quick_sort(std::vector<int>& arr, int low, int high) {
    if (low < high) {
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j] < pivot) {
                i++;
                std::swap(arr[i], arr[j]);
            }
        }
        std::swap(arr[i + 1], arr[high]);
        int pi = i + 1;
        quick_sort(arr, low, pi - 1);
        quick_sort(arr, pi + 1, high);
    }
}

int main() {
    int n;
    std::cin >> n;
    std::vector<int> sequence(n);
    for (int i = 0; i < n; i++) {
        std::cin >> sequence[i];
    }
    quick_sort(sequence, 0, n - 1);
    for (int i = 0; i < n; i++) {
        std::cout << sequence[i] << " ";
    }
    std::cout << std::endl;
    return 0;
}